src/store/journal/journal.selector.ts

```diff
@@ -86,6 +86,12 @@ export const getCompletedTests = (journalModel: JournalModel): SearchResultTestS
   return journalModel.completedTests;
 };
 
+export const getCompletedTestOutcome = (completedTests: SearchResultTestSchema[], appRef: string): ActivityCode => {
+  return completedTests
+    .find((test) => test.applicationReference === parseInt(appRef, 10))
+    .activityCode;
```

Comment:
Find will return undefined when not found, so we could do with a ? before accessing `.activityCode`;

Reply:
agreed.  will update